src/Blockcore.Indexer.Core/Storage/Mongo/MongoData.cs

```diff
@@ -383,22 +390,22 @@ public QueryResult<RichlistTable> Richlist(int offset, int limit)
          int total = (int)RichlistTable.Find(filter).CountDocuments();
 
          // If the offset is not set, or set to 0 implicit, we'll reverse the query and grab last page as oppose to first.
-         if (offset == 0)
-         {
-            // If limit is higher than total, simply use offset 0 and get all that exists.
-            if (limit > total)
-            {
-               offset = 1;
-            }
-            else
-            {
-               offset = (total - limit) + 1; // +1 to counteract the Skip -1 below.
-            }
-         }
+         //if (offset == 0)
```

Comment:
what if offset is -1?

Reply:
It cannot be any longer, the input parameters is validated in the API to be 0 or higher, or null.